### PR TITLE
handle invalid container names in "-m" to acorn run (#1087)

### PR DESCRIPTION
**Pull request by jsilverio22**
This PR handles when a user inputs a nonexistent container name in the `acorn run -m` flag it will now error out server side. 

Issue: #1087 
Signed-off-by: Joshua Silverio <joshua@acorn.io>

### CODE REVIEW

File: pkg/server/registry/apps/validator.go [hunk 1]
```diff
@@ -319,6 +325,19 @@ func checkMemory(memory v1.Memory, workloads map[string]v1.Container, specMemDef
...
 
+func validateMemoryRunFlags(memory v1.Memory, workloads map[string]v1.Container) *field.Error {
+	for key, value := range memory {
+		if key == "" {
+			continue
+		}
+		if _, ok := workloads[key]; !ok {
+			path := field.NewPath("spec", "memory")
+			return field.Invalid(path, fmt.Sprintf("%s=%v", key, *value), v1.ErrInvalidWorkload.Error())
+		}
```
Comment (tylerslaton) [inline pkg/server/registry/apps/validator.go:336]: nit: Simplifies the code a bit and uses the path pattern used in `checkMemory`
```suggestion
		if _, ok := workloads[key]; !ok {
			path := field.NewPath("spec", "memory", key)
			return field.Invalid(path, *value, v1.ErrInvalidWorkload.Error())
		}
```

File: pkg/server/registry/apps/validator.go [hunk 2]
```diff
@@ -319,6 +325,19 @@ func checkMemory(memory v1.Memory, workloads map[string]v1.Container, specMemDef
 	return validationErrors
 }
 
+func validateMemoryRunFlags(memory v1.Memory, workloads map[string]v1.Container) *field.Error {
```
Comment (tylerslaton) [inline pkg/server/registry/apps/validator.go:328]: Should this logic just be in `v1/memory.go` file as apart of `v1.ValidateMemory`?

**Review by tylerslaton — APPROVED** [review]
Two non-blocking comments.

File: pkg/server/registry/apps/validator.go (hunk 2 above)
Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go:328]: I thought about that but it seemed like v1.validatememory handled more of the quantity logic/validation so it made sense to me to separate them out open to changing to other way if you or others have a strong opinion

File: pkg/server/registry/apps/validator.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go:336]: any preference/thoughts?

File: pkg/server/registry/apps/validator.go (hunk 2 above)
Comment (tylerslaton) [reply pkg/server/registry/apps/validator.go:328]: Only real difference is that this same logic won't happen on the controller unless it exists in `v1.ValidateMemory`. Right now validation happens on both the controller and the API server.

File: pkg/server/registry/apps/validator.go (hunk 1 above)
Comment (StrongMonkey) [reply pkg/server/registry/apps/validator.go:336]: Should we mention in the error message that the container name doesn't exist/get defined the app? I think with the current error message, user still won't have an idea what's wrong the the specification.

Comment (thedadams) [reply pkg/server/registry/apps/validator.go:336]: Change `*value` to `key`, as: `return field.Invalid(path, key, v1.ErrInvalidWorkload.Error())`

File: pkg/server/registry/apps/validator.go (hunk 2 above)
Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go:328]: ahh okay in that case I will move to v1.validatememory

File: pkg/server/registry/apps/validator.go (hunk 1 above)
Comment (tylerslaton) [reply pkg/server/registry/apps/validator.go:336]: Yeah that's right. We want this to be as clear as possible that the user defined a workload to change that does not exist.

Comment (thedadams) [reply pkg/server/registry/apps/validator.go:336]: Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

**Review by thedadams — CHANGES_REQUESTED** [review]

File: pkg/apis/internal.acorn.io/v1/memory.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ var (
 	ErrInvalidAcornMemory   = errors.New("invalid memory from Acornfile")
 	ErrInvalidSetMemory     = errors.New("invalid memory set by user")
 	ErrInvalidDefaultMemory = errors.New("invalid memory default")
+	ErrInvalidWorkload      = errors.New("invalid workload name set by user")
```
Comment (StrongMonkey) [inline pkg/apis/internal.acorn.io/v1/memory.go:55]: Do we call containers as `workload` in acorn?

Comment (thedadams) [reply pkg/apis/internal.acorn.io/v1/memory.go:55]: Acorn containers and jobs are collectively referred to as workloads.

File: pkg/server/registry/apps/validator.go (hunk 2 above)
Comment (tylerslaton) [reply pkg/server/registry/apps/validator.go:328]: Discussed offline, but we'll go ahead and put this into an issue an track it as a follow-up.

File: pkg/server/registry/apps/validator.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go]: the error msg is now @thedadams @StrongMonkey 
```
 ERROR:  App.api.acorn.io "winter-lake" is invalid: spec.memory: Invalid value: "ng": workload name set by user does not exist
```

Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go]: > Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

It does to propagated up through `checkMemory` error  slice

**Review by StrongMonkey — APPROVED** [review]

File: pkg/server/registry/apps/validator.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/server/registry/apps/validator.go]: > Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

should be handled now for multiple errs

**Pull request merged**